src/auth/auth.controller.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Query,
+  HttpException,
+  Post,
+  UseGuards,
+  Req,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { TRequest } from 'src/common/types/user.types';
```

Comment:
PullProofAI

The `AuthController` class is missing necessary imports and decorators.
```typescript
1: import { Controller, Get, Query, Post, UseGuards, Req } from '@nestjs/common';
2: import { AuthService } from './auth.service';
3: import { ApiBearerAuth, ApiTags } from '@nestjs/swagger';
4: import { JwtAuthGuard } from './jwt-auth.guard';

@ApiTags('Auth')
@Controller()
export class AuthController {
    constructor(private readonly authService: AuthService) {}
```



<!-- This is an auto-generated comment by PullProofAI -->